contracts/SnapshotToken/Extensions/SimpleVote.sol

```diff
@@ -0,0 +1,119 @@
+pragma solidity 0.4.26;
+
+import "./ISnapshotableToken.sol";
+/* import "../../Math.sol"; */
+
+contract SimpleVote {
+
+    ////////////////////////
+    // Types
+    ////////////////////////
+
+    struct Proposal {
+        uint256 inFavor;
+        uint256 against;
+        uint256 endTime;
+        address owner;
+        uint256 snapshotId;
+        mapping (address => bool) hasVoted;
+
+    }
+
+    /////////////////////////
+    // Immutable state
+    ////////////////////////
+    uint256 private VOTING_PERIOD;
+    ISnapshotableToken private TOKEN;
+    // % of tokens to participate in a voting to make it valid
+    uint256 QUORUM = 40;
+
+    /////////////////////////
+    // Mutable state
+    ////////////////////////
+    Proposal[] private _proposals;
+    mapping (address => bool) private _hasActiveProposal;
+
+    /////////////////////////
+    // Events
+    ////////////////////////
+    event LogNewProposal(address indexed owner, uint256 indexed proposalIndex, uint256 snapshotId, uint256 endTime);
+    event LogProposalResult(uint256 indexed proposalIndex, bool hasPassed, uint256 inFavor, uint256 against);
+
+    ////////////////////////
+    // Constructor
+    ////////////////////////
+
+    /// @param votingPeriodInDays amount of days that each vote will take
+    /// @param token which balances are used for voting
+    constructor(
+        ISnapshotableToken token,
+        uint256 votingPeriodInDays
+    )
+        public
+    {
+        VOTING_PERIOD = votingPeriodInDays * 1 days;
+        TOKEN = token;
+    }
+
+    /////////////////////////
+    // Public functions
+    ////////////////////////
+
+    /// @dev Creates a proposal that will end VOTING_PERIOD days from now
+    function addProposal() public {
+        require(!_hasActiveProposal[msg.sender]);
+
+        uint256 sId = TOKEN.currentSnapshotId();
```

Comment:
the comment for createSnapshot() said not to rely on the returned integer, so I am doing two separate calls.